src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -579,6 +580,21 @@ public bool IsTupleCompatible()
         }
 
         /// <summary>
+        /// Verify if the given type is a tuple of a given cardinality, or can be used to back a tuple type 
+        /// with the given cardinality. 
+        /// </summary>
+        public bool IsTupleOrCompatibleWithTupleOfCardinality(int targetCardinality)
```

Comment:
Given our discussion about `IsTupleCompatible` (which answers yes for type that are already tuples), I would suggest the same name here (just different parameter).